From cc157ca5e6f6a83b3f463324bf6c84a7251c7859 Mon Sep 17 00:00:00 2001
From: Joachim Breitner <mail@joachim-breitner.de>
Date: Tue, 12 Jan 2010 11:45:16 +0100
Subject: Cast to (intptr_t) in PackedPtrSet.h

The check was (((int)(*vp)) & 01), now (((intptr_t)(*vp)) & 01). Should
be safe.
---
 sfworks/common/PackedPtrSet.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/sfworks/common/PackedPtrSet.h b/sfworks/common/PackedPtrSet.h
index 4687fbf..2b048ee 100644
--- a/sfworks/common/PackedPtrSet.h
+++ b/sfworks/common/PackedPtrSet.h
@@ -76,7 +76,7 @@ public:
         for(; *vp; ++vp) {
             if (*vp == p)
                 return false;
-            if (((int)(*vp)) & 01) {
+            if (((intptr_t)(*vp)) & 01) {
                 *vp = p;
                 return true;
             }
-- 
1.6.6

